core/systems/disks/steam_removable_media_manager.gd

```
@@ -0,0 +1,241 @@
+
+func _init() -> void:
+
+	# Check the required system files exist for steam_removable_media
+	if FileAccess.file_exists(SRM_FORMAT_MEDIA):
+		format_capable = true
+	if FileAccess.file_exists(SRM_INIT_MEDIA):
+		init_capable = true
+	if FileAccess.file_exists(STEAMOS_RETRIGGER_AUTOMOUNTS):
+		retrigger_capable = true
+
+	# These scripts are hard coded to mmcblk0
+	if FileAccess.file_exists(SDCARD_PATH):
+		if FileAccess.file_exists(STEAMOS_FORMAT_SDCARD):
+			format_sd_capable = true
+		if FileAccess.file_exists(STEAMOS_TRIM_DEVICES):
+			trim_capable = true
+
+	logger.debug("format_capable: " + str(format_capable))
+	logger.debug("init_capable: " + str(init_capable))
+	logger.debug("format_sd_capable: " + str(format_sd_capable))
+	logger.debug("retrigger_capable: " + str(retrigger_capable))
+	logger.debug("trim_capable: " + str(trim_capable))
+
+	# Get the current drives and ensure they are always up to date
+	udisks2.devices_updated.connect(update_drives)
+	update_drives()
+
+
+## Rebuilds the device_tree with the current device list
+func update_drives(devices: Array[BlockDevice] = []):
+	if devices == []:
+		logger.debug("Got empty array, grabbing devices from udisks2")
+		devices = udisks2.get_devices()
+	self.device_tree = []
+	for block in devices:
+		if not _block_device_has_protected_mount(block):
+			logger.debug(block.dbus_path, "has no protected mounts.")
+			device_tree.append(block)
+	self.drives_updated.emit(self.device_tree)
+
+
+## Calls the SteamRemovableMedia format-media script to format a drive as EXT4
+## and intialize it as a steam library
+func format_drive(device: BlockDevice) -> void:
+	if not format_capable:
+		logger.error("System is not format capable")
+		return
+
+	# This should never hit if using our device tree, but  the method is public so
+	# make sure.
+	if _block_device_has_protected_mount(device):
+		logger.error("Attempted to format device with protected mount. Illegal Operation.")
+		return
```

We should also return a custom error code here too.